source/common/stats/stat_data_allocator_impl.h

```diff
@@ -167,6 +169,73 @@ class NullGaugeImpl : public Gauge {
   uint64_t value() const override { return 0; }
 };
 
+/**
+ * TextReadout implementation that wraps a StatData. Max length 15; set() will truncate.
```

Comment:
per f2f conversation this is really cool but if we wait till your hot-restart-without-shm PR then this can be simpler and not have limits.

Does it make sense to wait till then?

Reply:
also during an update an observer might see garbage for the string the atomic behavior is per-uint64. In fact the length might be adjusted and expose old data, which is technically a security hole, though you only get 15 bytes of data through :)